### PR TITLE
use PaymentTypeFilter as list

### LINKED ISSUE #699: payment type filter to include ClosedChannels type
ref https://github.com/breez/c-breez/issues/697 and https://github.com/breez/breez-sdk/pull/521

### PR DESCRIPTION
**Pull request by ubbabeck**
fixes #699

### CODE REVIEW

File: lib/bloc/account/account_bloc.dart [hunk 1]
```diff
@@ -247,14 +247,14 @@ class AccountBloc extends Cubit<AccountState> with HydratedMixin {
   }
 
   void changePaymentFilter({
-    sdk.PaymentTypeFilter? filter,
+    List<sdk.PaymentTypeFilter>? filter,
```
Comment (erdemyerebasmaz) [inline lib/bloc/account/account_bloc.dart:250]: rename variable to `filters`

File: lib/bloc/csv_exporter.dart [hunk 2]
```diff
@@ -114,9 +114,9 @@ class CsvExporter {
 
   String _appendFilterInformation(String filePath) {
     _log.info("add filter information to path started");
-    if (filter == PaymentTypeFilter.Sent) {
+    if (!filter.contains(PaymentTypeFilter.Received)) {
       filePath += "_sent";
-    } else if (filter == PaymentTypeFilter.Received) {
+    } else if (!filter.contains(PaymentTypeFilter.Sent)) {
       filePath += "_received";
     }
```
Comment (erdemyerebasmaz) [inline lib/bloc/csv_exporter.dart:121]: Think this logic should be changed to sth like:

```
    List<PaymentTypeFilter>? filterList = filters;
    if (filterList != null && filterList != PaymentTypeFilter.values) {
      for (var filter in filterList) {
        switch (filter) {
          case PaymentTypeFilter.Sent:
            filePath += "_sent";
            break;
          case PaymentTypeFilter.Received:
            filePath += "_received";
            break;
          case PaymentTypeFilter.ClosedChannels:
            filePath += "_closed_channels";
            break;
        }
      }
    }
```

Where filter information is appended unless `filters` is empty or null(_unfiltered_), or filters has all `PaymentTypeFilter.values` (_unfiltered_)

File: lib/routes/home/account_page.dart [hunk 3]
```diff
@@ -31,8 +31,8 @@ class AccountPage extends StatelessWidget {
   const AccountPage(
     this.firstPaymentItemKey,
     this.scrollController, {
-    super.key,
-  });
+    Key? key,
+  }) : super(key: key);
```
Comment (erdemyerebasmaz) [inline lib/routes/home/account_page.dart:35]: Please revert this and upgrade your Flutter to latest stable.

File: lib/routes/home/account_page.dart [hunk 4]
```diff
@@ -69,8 +69,8 @@ class AccountPage extends StatelessWidget {
       if (fromTimestamp != null && toTimestamp != null) {
         return fromTimestamp < milliseconds && milliseconds < toTimestamp;
       }
-      if (paymentFilters.filter != PaymentTypeFilter.All) {
-        return paymentMinutiae.paymentType.name == paymentFilters.filter.name;
+      if (paymentFilters.filters != PaymentTypeFilter.values) {
```
Comment (erdemyerebasmaz) [inline lib/routes/home/account_page.dart:72]: We also need to check if filters is empty.

```
      if ((paymentFilters.filters == null) || (paymentFilters.filters != null && paymentFilters.filters! != PaymentTypeFilter.values)) {
```

File: lib/routes/home/widgets/payments_filter/payment_filter_exporter.dart [hunk 5]
```diff
@@ -14,7 +14,7 @@ import 'package:share_plus/share_plus.dart';
 
 class PaymentmentFilterExporter extends StatelessWidget {
   final _log = Logger("PaymentmentFilterExporter");
-  final PaymentTypeFilter filter;
+  final List<PaymentTypeFilter> filter;
```
Comment (erdemyerebasmaz) [inline lib/routes/home/widgets/payments_filter/payment_filter_exporter.dart:17]: Make optional and rename to filters
```suggestion
  final List<PaymentTypeFilter>? filters;
```

File: lib/bloc/account/account_bloc.dart (hunk 1 above)
Comment (ubbabeck) [reply lib/bloc/account/account_bloc.dart]: Added

File: lib/bloc/csv_exporter.dart [hunk 6]
```diff
@@ -17,7 +17,7 @@ class CsvExporter {
   final AccountState accountState;
   final bool usesUtcTime;
   final String fiatCurrency;
-  final PaymentTypeFilter filter;
+  final List<PaymentTypeFilter> filter;
```
[the remark this answers is not in the document]
Comment (ubbabeck) [reply lib/bloc/csv_exporter.dart]: Added

File: lib/bloc/csv_exporter.dart (hunk 2 above)
Comment (ubbabeck) [reply lib/bloc/csv_exporter.dart:130]: Thanks, I've added this filtering, but added closed channels to the sent case.

File: lib/routes/home/widgets/payments_filter/payment_filter_exporter.dart (hunk 5 above)
Comment (ubbabeck) [reply lib/routes/home/widgets/payments_filter/payment_filter_exporter.dart]: Added

**Comment by erdemyerebasmaz** [comment]
> @erdemyerebasmaz When testing I discovered that export to csv does not filter properly. Should we handle this in this pr or open a separate issue for it?

@ubbabeck Let's handle it on this PR.

**Comment by ubbabeck** [comment]
After discussing with @erdemyerebasmaz we agreed to merge this PR and open a new issue for handling the filtering in the UI. 

The reasoning for this is that the changes implemented with the closing channel filter includes breaking changes, so to allow others to also build using master of the sdk we're merging this PR.

**Pull request merged**